Terminal.Gui/Clipboard/Clipboard.cs

```
@@ -1,174 +1,178 @@
-	/// <returns><see langword="true"/> the OS clipboard was retrieved, <see langword="false"/> otherwise.</returns>
-	public static bool TryGetClipboardData (out string result)
-	{
-		if (IsSupported && Application.Driver.Clipboard.TryGetClipboardData (out result)) {
-			if (_contents != result) {
-				_contents = result;
-			}
-			return true;
-		}
-		result = string.Empty;
-		return false;
-	}
-
-	/// <summary>
-	/// Pastes the <paramref name="text"/> to the OS clipboard if possible.
-	/// </summary>
-	/// <param name="text">The text to paste to the OS clipboard.</param>
-	/// <returns><see langword="true"/> the OS clipboard was set, <see langword="false"/> otherwise.</returns>
-	public static bool TrySetClipboardData (string text)
-	{
-		if (IsSupported && Application.Driver.Clipboard.TrySetClipboardData (text)) {
-			_contents = text;
-			return true;
-		}
-		return false;
-	}
+    private static string _contents = string.Empty;
+
+    /// <summary>Returns true if the environmental dependencies are in place to interact with the OS clipboard.</summary>
+    /// <remarks></remarks>
+    public static bool IsSupported => Application.Driver.Clipboard.IsSupported;
+
+    /// <summary>Gets (copies from) or sets (pastes to) the contents of the OS clipboard.</summary>
+    public static string Contents {
+        get {
+            try {
+                if (IsSupported) {
+                    string clipData = Application.Driver.Clipboard.GetClipboardData ();
+                    if (clipData == null) {
+                        // throw new InvalidOperationException ($"{Application.Driver.GetType ().Name}.GetClipboardData returned null instead of string.Empty");
+                        clipData = string.Empty;
+                    }
+
+                    _contents = clipData;
+                }
+            }
+            catch (Exception) {
+                _contents = string.Empty;
```

Out of scope for this PR, but this makes me 🤨 at a cursory glance.

---

Also, I've been thinking for a while now that we should put hooks for logging in the library, especially around exception handlers, and _especially_ those that eat exceptions.

We don't have to actually write the logs - it can be done via the standard logging infrastructure in modern .net, so anyone who wants logging can provide whatever logging implementation they wish.